packages/components/src/site-thumbnail/index.tsx

```diff
@@ -1,54 +1,82 @@
+import { css, CSSObject } from '@emotion/css';
 import classnames from 'classnames';
 import { ReactNode } from 'react';
 import './style.scss';
 import { MShotsOptions, useMshotsImg } from './use-mshots-img';
 import { getTextColorFromBackground } from './utils';
 
-const MSHOTS_OPTION: MShotsOptions = {
-	vpw: 1200,
-	vph: 1200,
-	w: 374 * 2,
-};
+export type SizeCss = { width: number; height: number };
+const ASPECT_RATIO = 16 / 11;
+export const SITE_THUMBNAIL_DIMENSIONS = {
+	small: {
+		width: 108,
+		height: 78,
+	},
+	medium: {
+		width: 401,
+		height: 401 / ASPECT_RATIO,
+	},
+} as const;
+
+const VIEWPORT_BASE = 1200;
 
 type Props = {
 	backgroundColor?: string;
-	className?: string;
+	style?: CSSObject;
 	mShotsUrl?: string;
-	size?: 'small' | 'medium';
+	dimension?: SizeCss;
+	dimensionsSrcset?: Array< SizeCss >;
+	sizesAttr?: string;
 	children?: ReactNode;
 	alt?: string;
 	bgColorImgUrl?: string;
+	viewport?: number;
 	mshotsOption?: MShotsOptions;
 };
 
 export const SiteThumbnail = ( {
 	backgroundColor,
 	children,
-	className,
+	style,
 	alt,
 	mShotsUrl = '',
 	bgColorImgUrl,
-	size = 'small',
-	mshotsOption = MSHOTS_OPTION,
+	dimension = SITE_THUMBNAIL_DIMENSIONS.small,
+	dimensionsSrcset = [],
+	sizesAttr = '',
+	viewport = VIEWPORT_BASE,
+	mshotsOption,
 }: Props ) => {
-	const { src, isLoading, isError, imgRef } = useMshotsImg( mShotsUrl, mshotsOption );
+	const options: MShotsOptions = {
+		vpw: viewport,
+		vph: viewport,
+		w: dimension.width,
+		h: dimension.height,
+		...mshotsOption,
+	};
+	const { imgProps, isLoading, isError, imgRef } = useMshotsImg( mShotsUrl, options, [
+		...dimensionsSrcset,
+		dimension,
+	] );
 
 	const color = backgroundColor && getTextColorFromBackground( backgroundColor );
 
 	const classes = classnames(
 		'site-thumbnail',
-		className,
 		isLoading ? 'site-thumbnail-loading' : 'site-thumbnail-visible',
-		`site-thumbnail__size-${ size }`
+		// default image width, height given by dimension if not specified in style
+		css( dimension, style )
 	);
 
 	const showLoader = mShotsUrl && ! isError;
 
+	const blurSize = dimension.width >= SITE_THUMBNAIL_DIMENSIONS.medium.width ? 'large' : 'small';
```

Comment:
Nice!

Did we get a large blur in, though? The Grid view doesn't seem blurred to me:

<img width="443" alt="image" src="https://user-images.githubusercontent.com/36432/187281152-d9edcf9c-ec41-4659-a228-74792467abee.png">

Reply:
Fixed!, Copilot suggested 'large' instead of 'medium'.